### PR TITLE
:sparkles: Rail fence cipher

### PR DESCRIPTION
**Pull request by Sigurdurhelga**
added rail fence cipher

### CODE REVIEW

File: encryptor/encryptors/railfence.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    FILLER_CHARACTER = '.'
+    DECRYPTION_POINTER = '#'
+
+    @staticmethod
+    def encrypt(text: str, rail_height: int = DEFAULT_RAILS):
+        try:
+            rail_height = int(rail_height)
+        except ValueError:
+            raise WrongTypeParameter("rail_height", int, type(rail_height))
+        """
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/railfence.py]: Please, add the docstring above your `try except` block

File: encryptor/encryptors/railfence.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        to make railfence work as in most examples we must first clear the
+        text of all extra stuff (special characters, etc)
+        """
+
+        new_text = []
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/railfence.py]: You might do the lines 22-26 with a list comprehension:
```python3
  text = ''.join([c for c in text if c.isalnum() or c == ' '])
```

File: tests/api/api_railfence_test.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def test_railfence_decrypt_api(client):
+    response = client.get(
+        "/api/railfence/decrypt/hloolelwrd?rail_height=2").get_json()
+    assert response["status"] == 200
+    assert response["content"] == "helloworld"
+
+
+def test_ceasar_decrypt_api_no_offset(client):
```
Comment (rafaeelaudibert) [inline tests/api/api_railfence_test.py]: Please rename it to `test_railfence_decrypt_api_no_rail_height`

File: tests/api/api_railfence_test.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def test_railfence_encrypt_api(client):
+    response = client.get(
+        "/api/railfence/encrypt/helloworld?rail_height=2").get_json()
+    assert response["status"] == 200
+    assert response["content"] == "hloolelwrd"
+
+
+def test_railfence_encrypt_api_no_offset(client):
```
Comment (rafaeelaudibert) [inline tests/api/api_railfence_test.py]: Please rename it to `test_railfence_encrypt_api_no_rail_height`

File: tests/api/api_railfence_test.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import pytest
```
Comment (rafaeelaudibert) [inline tests/api/api_railfence_test.py]: Could you please add a test when you pass an invalid rail height? Such as a string, expecting a 400 error as return

File: tests/encryptors/railfence_test.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+
+import pytest
+
+from encryptor.encryptors import RailFence
+from encryptor.errors import *
+
```
Comment (rafaeelaudibert) [inline tests/encryptors/railfence_test.py]: Could you please add a test, to test the default rail height?

Comment (Sigurdurhelga) [reply tests/encryptors/railfence_test.py]: I believe those have already been implemented see `test_railfence_encrypt_api_no_rail_height` & `test_railfence_decrypt_api_no_rail_height`

Comment (rafaeelaudibert) [reply tests/encryptors/railfence_test.py]: They have been implemented in the API. But the unit test (the one which we do for the class alone, without integrations with the rest of the system) is missing

**Comment by Sigurdurhelga** [comment]
I think that's all of it

**Comment by rafaeelaudibert** [comment]
I think that's it!
Thanks for contributing to the project!

Enjoy your Hacktober, and keep contributing to OSS.
✨ 🎉

**Pull request merged**